[ezeAng] iP

ezeAng opened the pull request:
# C4PO Pro (Student Edition)

![C4PO](https://static.thenounproject.com/png/1454734-200.png)

> "For a student, you seem to do an incessant amount of thinking."
>  
>   -- C4PO, 'NUS Wars: Episode II- Attack Of The Students'.

C4PO frees your mind from having to keep track of your tasks to do.
C4PO is:
- text-based
- easy to learn
- ~~FAST~~ EXTREMELY FAST to use

All you need to do is,
1) download it from [here](https://pages.github.com/).
2) double-click it.
3) add your tasks.
4) let it manage your tasks for you 😉

And it is **FREE**!

Features:
- [X] Managing Tasks
- [X] Managing Deadlines
- [ ] Reminders (coming soon)

Just a sneak peak at how the `main` method works:
```java
public static void main(String[] args) {
        new C4PO(FILE_P).run();
}
```

Commit:
Add Gradle support

ezeAng committed:
Level-1

ezeAng committed:
Level-2

ezeAng committed:
Level-3

ezeAng committed:
Level-3.1 Task class

ezeAng committed:
Level-4

ezeAng committed:
Level-5

ezeAng committed:
Level-6

ezeAng committed:
A-TextUiTesting

ezeAng committed:
branch-Level-7

ezeAng committed:
branch-Level-8

ezeAng committed:
Merge tag 'branch-Level-7'

Level-7 branch

ezeAng committed:
fixed merge conflicts

ezeAng committed:
A-MoreOOP

ezeAng committed:
Packaged all classes except C4PO public class

ezeAng committed:
Merge remote-tracking branch 'origin/add-gradle-support'

ezeAng committed:
Recommit

ezeAng committed:
Add JUnit tests
tests

ezeAng committed:
Fix deadline and storage printing introduction and loading bug

ezeAng committed:
Add javaDoc comments

ezeAng committed:
Follow coding standard

ezeAng committed:
Create new feature Find

ezeAng committed:
Merge branch 'branch-Level-9'

ezeAng committed:
Add GUI to C4PO(Duke)

zichen-3974 commented on src/main/java/C4PO.java:
```diff
@@ -0,0 +1,111 @@
+
+import src.main.c4po.*;
```
Perhaps you would like to list import classes explicitly?

zichen-3974 commented on src/main/java/C4PO.java:
```diff
@@ -0,0 +1,111 @@
+     * You should have your own function to generate a response to user input.
+     * Replace this stub with your completed method.
+     */
+    protected String getResponse(String input) throws BotException {
```
Perhaps you would like to add Javadocs for this method

CarrieLi1015 commented on src/main/java/src/main/c4po/Ui.java:
```diff
@@ -88,6 +130,19 @@ public static void showIntroduction() {
+    /**
+     * Returns String of welcomes messages to the output terminal
+     */
+    public static String showIntroduction(boolean b) {
```
Might consider to have a more meaning name for the boolean variable here.

zichen-3974 commented on src/main/java/src/main/c4po/AddTaskCommand.java:
```diff
@@ -0,0 +1,77 @@
+package src.main.c4po;
```
Perhaps you can consider putting all *Command classes into a package for commands

CarrieLi1015 reviewed (commented):
Overall nice work! Just remember to follow the Java doc standard. For example, description of param/return/throw should begin with upper case and end with "." 👍

zichen-3974 commented on src/main/java/src/main/c4po/DeleteCommand.java:
```diff
@@ -0,0 +1,74 @@
+     * Executes the command with actions specific to each extension of this
+     * class Command
+     *
+     * @param tasks          are the list of tasks
+     * @param ui             is the instance of UI
+     * @param storage        the instance of Storage which holds and writes to the data file
+     * @param isStringOutput
```
Your descriptions are nice and clear, though you may want to take note of punctuation and begin with uppercase for each parameter description.

zichen-3974 commented on src/main/java/src/main/c4po/DeleteCommand.java:
```diff
@@ -0,0 +1,74 @@
+        try {
+            Task toDelete = tasks.getItem(toDeletePosition);
+
+            boolean isSuccess = tasks.deleteItem(toDelete);
```
I like how you name your variables meaningfully 👍

zichen-3974 commented on src/main/java/src/main/c4po/Parser.java:
```diff
@@ -0,0 +1,102 @@
+            return new QuitCommand();
+        } else if (receive.length() > 3 && "list".equalsIgnoreCase(receive.strip())) {
+            return new ListCommand();
+        } else if (receive.length() > 3 && "mark".equalsIgnoreCase(receive.substring(0,4))) {
```
Nice implementation, perhaps you can consider other ways to do this too? e.g. using split(),  switch statements.

zichen-3974 reviewed (commented):
I think you did a nice job overall! Just some very minor issues with your Javadocs :)

ezeAng committed:
Add assertions to parts of the code.

There are serveral parts where assertions were used. These are as follows:
UnmarkCommand
MarkCommand
TaskList

They check that the index number provided is >0, in other words, >= 1

ezeAng committed:
Add Assertions to parts of code

Assertions are needed to check that the value persisted downwards
into other deeper functions are not wrong

Assertions to add:
- index value - check that index input to the commands:
*MarkCommand
*UnmarkCommand
Tasklist.get__ commands

Index values are 0-indexed usually, but the user input index starts from the
convention from 1. We want to make sure the value is not equal to 0 by the time
it reaches the deeper nested functions, which account for the 1-index by subtracting
the received index by 1.

ezeAng committed:
Merge pull request #1 from ezeAng/branch-A-Assertions

Branch A assertions

ezeAng committed:
dummy PR

ezeAng committed:
dummy commit for PR practice

ezeAng committed:
Merge pull request #2 from ezeAng/branch-A-CodingStandard

dummy PR

ezeAng committed:
Create Priority sorting functionality

To allow users to add a priority to their tasks

Since the original list is already sorted by chronological order (in order of creation)
we need another way to sort the tasks to see the most important tasks

Here we shall define a new command class - GetPrioritySortedCommand
which when executed, will poll a priority Queue, sorting the tasks
and returning a temporary array

ezeAng committed:
Merge branch 'branch-C-SortByTimeCreated'

ezeAng committed:
Add userguide, bug fixes and title name

ezeAng committed:
Add Ui.png
